File: Clase18/js/funciones.js
// Introducción a funciones en JS
// Tiempos
// - 1° Declaración o definición
function saludar() {
    let usuario = "Christian";
    let frase = `Hola, soy ${usuario}`;
    console.log(frase);
}

// - 2° Invocación o llamada
saludar();

// Parámetros
function saludarPersona(nombre) {
    let frase = `Hola, soy ${nombre}`;
    console.log(frase);
}

saludarPersona('Ariel');
saludarPersona(); // undefined

File: Clase27/funciones.py
# Funciones

# Tiempos de la función: 1° definición 2° invocación
# def saludar():
#     print("Hola")
    
# saludar()

# Parámetros
# def saludar(nombre):
#     print(f"Hola {nombre}")
    
# saludar(["a", "b", "c"])
# saludar("Christian")

# Parámetros y retorno
def saludar(nombre): # el parámetro es de ámbito local
    return f"Hola {nombre}"

saludo = saludar("Christian") # "Hola Christian"

# Ámbitos
# print(nombre) # NameError: name 'nombre' is not defined